app/controllers/users_controller.rb

```diff
@@ -20,6 +20,13 @@ def create
   def show
   end
 
+  def logout
+    session.delete(:cart)
```

Comment:
I think you can delete lines 24 and 26, and replace it with reset_session. Here's an API doc about it: https://apidock.com/rails/ActionController/Base/reset_session

Reply:
Let's talk about this tomorrow. The reset_session wasn't allowing the flash message to work.